code/__HELPERS/spatial_info.dm

```diff
@@ -218,9 +218,15 @@
 /// Modifies input list.
 /proc/mob_only_listeners(list/atom/movable/hearers)
 	RETURN_TYPE(/list/mob)
-	for(var/hearer_index in 1 to hearers.len)
+
+	var/hearers_length = hearers.len
+	if(isnull(hearers) || !hearers_length)
```

Comment:
This will throw an error if hearers is null.

Reply:
I put it back, just add suggestion if you decide to go for better readability over saving a few ops